datalad/core/local/tests/test_run.py

```diff
@@ -571,3 +571,40 @@ def test_run_remove_keeps_leading_directory(path):
                result_renderer=None),
         action="run.remove", status="ok")
     assert_repo_status(ds.path)
+
+
+@with_tree(tree={"foo": "f", "bar": "b"})
+def test_dry_run(path):
```

Comment:
I wonder if worthwhile adding into a test some uninstalled subdataset with run having input and/or output path(s) pointing to it just to ensure that `--dry-run` (or some logic above) doesn't blow somehow due to absent (since subdataset is uninstalled) files, and that the subdataset does not get installed (according to the code it shouldn't but that would be the point of the test to ensure that it will not ;-))

Reply:
Indeed.  Thanks for bring up subdatasets.  I overlooked them when working on this.